cirq/__init__.py

```diff
@@ -331,6 +331,7 @@
     dirac_notation,
     eye_tensor,
     fidelity,
+    infer_qid_shape,
```

Comment:
I'm not convinced yet that this needs to be exposed to the cirq namespace. 
It feels like this would be useful mostly for other functions in the qis package.

Reply:
We've never really held back from exposing every public function to the global namespace before. We've never had a good packaging strategy before either. But in any event, we should make the criteria clear for global namespace instead of picking and choosing per PR

Reply:
I'm always in favor of making things package-private and expanding as necessary

Reply:
I've removed the top-level exposure.